src/main/java/de/tum/in/www1/artemis/repository/TextExerciseRepository.java

```diff
@@ -36,7 +37,18 @@ public interface TextExerciseRepository extends JpaRepository<TextExercise, Long
     @EntityGraph(type = LOAD, attributePaths = { "teamAssignmentConfig", "categories", "competencies", "plagiarismDetectionConfig" })
     Optional<TextExercise> findWithEagerTeamAssignmentConfigAndCategoriesAndCompetenciesAndPlagiarismDetectionConfigById(Long exerciseId);
 
-    @Query("select textExercise from TextExercise textExercise left join fetch textExercise.exampleSubmissions exampleSubmissions left join fetch exampleSubmissions.submission submission left join fetch submission.results result left join fetch result.feedbacks left join fetch submission.blocks left join fetch result.assessor left join fetch textExercise.teamAssignmentConfig where textExercise.id = :#{#exerciseId}")
+    @Query("""
```

Comment:
Can derive this query and use an entity graph to fetch data?

Reply:
As a matter of fact, it's not possible, because EntityGraph doesn't support child class fields which `submission.blocks` is one of.